server/src/main/java/io/deephaven/server/arrow/ArrowFlightUtil.java

```diff
@@ -716,6 +861,8 @@ public synchronized void close() {
                 if (preExportSubscriptions != null) {
                     preExportSubscriptions = null;
                 }
+
+                listener.onCompleted();
```

Comment:
We probably want to do this only when `bmp == null`. The `bmp` will call `onCompleted` during the update propagation job after having invoked `removeSubscription`. 

Also, should we wrap this call in a `GrpcUtil.safelyExecute`?

Note that if we call `onCompleted` after having passed the listener to the bmp, then `onCompleted` must be called `GrpcUtil.safelyExecuteLocked(listener, listener::onCompleted)`. The reason is that the observers are not thread safe and you can end up deadlocking in gRPC land (the bmp might be in the middle of sending another update before actually removing the subscription).

Reply:
I made this change, but please verify it looks correct.